### PR TITLE
mrc-3521 remove/rename archive

### PR DESCRIPTION
**Pull request by hillalex**
Allow removing or renaming the file archive

### CODE REVIEW

File: R/config.R [hunk 1]
```diff
@@ -101,6 +102,42 @@ config_set_use_file_store <- function(value, root) {
...
+    if (!config$core$use_file_store) {
+      stop("if 'path_archive' is NULL, then 'use_file_store' must be TRUE")
+    }
+    path_archive <- file.path(root$path, config$core$path_archive)
+    if (fs::dir_exists(path_archive)) {
+      fs::dir_delete(file.path(root$path, config$core$path_archive))
+    }
+    config$core["path_archive"] <- list(NULL)
+  } else {
+    if (!is.null(config$core$path_archive)) {
```
Comment (richfitz) [inline R/config.R]: move this up to be `else if`?

File: R/config.R [hunk 2]
```diff
@@ -101,6 +102,42 @@ config_set_use_file_store <- function(value, root) {
...
+    return()
+  }
+
+  if (is.null(value)) {
+    if (!config$core$use_file_store) {
+      stop("if 'path_archive' is NULL, then 'use_file_store' must be TRUE")
+    }
+    path_archive <- file.path(root$path, config$core$path_archive)
+    if (fs::dir_exists(path_archive)) {
+      fs::dir_delete(file.path(root$path, config$core$path_archive))
```
Comment (richfitz) [inline R/config.R]:
```suggestion
      fs::dir_delete(path_archive)
```

File: R/config.R [hunk 3]
```diff
@@ -101,6 +102,42 @@ config_set_use_file_store <- function(value, root) {
...
+    }
+    config$core["path_archive"] <- list(NULL)
```
Comment (richfitz) [inline R/config.R:121]: this is super annoying, isn't it

File: R/config.R [hunk 4]
```diff
@@ -101,6 +102,42 @@ config_set_use_file_store <- function(value, root) {
...
+  config <- root$config
+
+  if (identical(value, config$core$path_archive)) {
+    message("'core.path_archive' was unchanged")
+    return()
+  }
+
+  if (is.null(value)) {
+    if (!config$core$use_file_store) {
+      stop("if 'path_archive' is NULL, then 'use_file_store' must be TRUE")
```
Comment (richfitz) [inline R/config.R]: At some point we should formalise what our error styling is. I feel we use initial capitals most of the time, but am not sure...

**Review by richfitz — CHANGES_REQUESTED** [review]
This looks great, some minor comments

File: R/config.R (hunk 3 above)
Comment (hillalex) [reply R/config.R:121]: yeah! took me a minute to figure this out

File: R/config.R (hunk 4 above)
Comment (hillalex) [reply R/config.R]: Ah yeah, this is inconsistent even just in this file. I'll change all of the messages in this file to start with uppercase

**Review by richfitz — APPROVED** [review]

**Pull request merged**